### ISSUE TITLE
bug: insufficient task parallelism with emergency picker enabled

### ISSUE THREAD

**Issue by hzxa21** [risingwavelabs/risingwave#15496]
### Describe the bug

When there are insufficient compactors in the cluster, it is likely that the compaction emergency picker will be triggered to prioritize L0 compaction tasks. However, we have seen in some users' cluster that the non-overlapping L0 level is too small to trigger intro-L0 compaction while base level is too big due to prioritizing L0 to base tasks in emergency picker. All these combined, we have seen that the in a cluster, there are 16 compactor cores available but only 2 running tasks even though the LSM tree is imbalanced and L0 files are piled up.

**Comment by Li0k** [comment]
related-issue https://github.com/risingwavelabs/risingwave/issues/14034

**Comment by Li0k** [comment]
discuss offline with @hzxa21.
The design purpose of the emergency picker is to ensure that compaction can run stably during the write-stop without being restricted by the validator, thereby avoiding deadlock caused by some bugs. However, we directly rely on the emergency picker to eliminate the accumulation of l0. This makes the emergency gradually bloated. Moreover, it destroys the original compaction priority, causing various problems.
- The sst accumulation problem of l0 has spread to lbase
- Resources cannot be fully utilized due to accumulation

In short, we are misusing it as a solution for eliminating l0 stacking rather than avoiding deadlock.

Therefore, I plan to reshape this setting. In most cases, follow the default compaction selection strategy, and only try to trigger it when the task cannot be selected and is in write-stop.
